src/hotspot/share/services/memReporter.hpp

```diff
@@ -83,10 +83,17 @@ class MemReporterBase : public StackObj {
 
     size_t amount = s1 - s2;
     assert(amount <= SIZE_MAX - _scale / 2, "size_t overflow");
-    amount = (amount + _scale / 2) / _scale;
-    // We assume the valid range for deltas [INT64_MIN, INT64_MAX] to simplify the code.
-    assert((sizeof(size_t) < sizeof(int64_t)) || (sizeof(size_t) == sizeof(int64_t) && amount <= INT64_MAX), "cannot fit scaled diff into size_t");
-    return (is_negative) ? -(int64_t)amount : (int64_t)amount;
+    size_t scaled = (amount / _scale);
+    if ((amount % _scale) > (_scale - 1)/2) {
+      scaled += 1;
+    }
+
+    if (scaled == 0) {
+      return 0;
+    }
+    assert(scaled <= INT64_MAX, "overflow");
```

Comment:
It looks like you have decided to support values in range [-INT64_MAX, INT64_MAX] not in [INT64_MIN, INT64_MAX]. I am OK with this.
If I am correct, could you please comment your decision in the code?